samcli/lib/providers/sam_stack_provider.py

```diff
@@ -0,0 +1,196 @@
+                return f
+
+        return None
+
+    def get_all(self) -> Generator[BuildableStack, None, None]:
+        """
+        Yields all the applications available in the SAM Template.
+        :yields Application: map containing the application information
+        """
+
+        for _, stack in self._stacks.items():
+            yield stack
+
+    def _extract_stacks(self) -> Dict[str, BuildableStack]:
+        """
+        Extracts and returns nested application information from the given dictionary of SAM/CloudFormation resources.
+        This method supports applications defined with AWS::Serverless::Application
+        :return dict(string : application): Dictionary of application LogicalId to the
+            Application object
+        """
+
+        result: Dict[str, BuildableStack] = {}
+
+        for name, resource in self._resources.items():
+
+            resource_type = resource.get("Type")
+            resource_properties = resource.get("Properties", {})
+            resource_metadata = resource.get("Metadata", None)
+            # Add extra metadata information to properties under a separate field.
+            if resource_metadata:
+                resource_properties["Metadata"] = resource_metadata
+
+            stack: Optional[BuildableStack] = None
+            if resource_type == SamBuildableStackProvider.SERVERLESS_APPLICATION:
+                stack = SamBuildableStackProvider._convert_sam_application_resource(
+                    self._stack_path, name, resource_properties
+                )
+            if resource_type == SamBuildableStackProvider.CLOUDFORMATION_STACK:
+                stack = SamBuildableStackProvider._convert_cfn_stack_resource(
+                    self._stack_path, name, resource_properties
+                )
+
+            if stack:
+                result[name] = stack
+
+            # We don't care about other resource types. Just ignore them
+
+        return result
+
+    @staticmethod
+    def _convert_sam_application_resource(
+        stack_path: str, name: str, resource_properties: Dict
+    ) -> Optional[BuildableStack]:
+        location = resource_properties.get("Location")
+
+        if isinstance(location, dict):
+            LOG.warning(
+                "Nested application '%s' has specified an application published to the "
+                "AWS Serverless Application Repository which is unsupported. "
+                "Skipping resources inside this nested application.",
+                name,
+            )
+            return None
+
+        location = cast(str, location)
+        if SamBuildableStackProvider.is_remote_url(location):
+            LOG.warning(
+                "Nested application '%s' has specified S3 location for Location which is unsupported. "
+                "Skipping resources inside this nested application.",
+                name,
+            )
+            return None
+        if location.startswith("file://"):
```

Comment:
Can users just specify string which is not starting with `file://`?

Reply:
yes, local absolute or relative path also works here.

According to the doc:

> Template URL, file path, or location object of a nested application.
https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-application.html#sam-application-location